tools/cli/domainCommands_test.go

```diff
@@ -0,0 +1,136 @@
+	"github.com/stretchr/testify/suite"
+	"github.com/urfave/cli"
+
+	"github.com/uber/cadence/client/frontend"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/domain"
+	"github.com/uber/cadence/common/types"
+)
+
+type domainCommandsSuite struct {
+	suite.Suite
+	mockCtrl             *gomock.Controller
+	serverFrontendClient *frontend.MockClient
+	domainHandler        *domain.MockHandler
+	cliCtx               *cli.Context
+	domainCLI            *domainCLIImpl
+}
+
+func TestDomainCommandsSuite(t *testing.T) {
+	s := new(domainCommandsSuite)
+	suite.Run(t, s)
+}
+
+func (s *domainCommandsSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+
+	s.serverFrontendClient = frontend.NewMockClient(s.mockCtrl)
+	s.domainHandler = domain.NewMockHandler(s.mockCtrl)
+
+	set := flag.NewFlagSet("test", 0)
+	set.Int(FlagContextTimeout, 10, "test flag")
+	s.cliCtx = cli.NewContext(nil, set, nil)
+	s.domainCLI = &domainCLIImpl{
+		s.serverFrontendClient,
+		s.domainHandler,
+	}
+}
+
+func (s *domainCommandsSuite) TearDownTest() {
+	s.mockCtrl.Finish() // assert mock’s expectations
+}
+
+func (s *domainCommandsSuite) TestRebalanceDomains_Success() {
+	listDomainsResp := getListDomainsResponse()
+	preferredCluster := "active"
+	s.serverFrontendClient.EXPECT().ListDomains(gomock.Any(), gomock.Any()).Return(listDomainsResp, nil)
+	expectedUpdateRequest := &types.UpdateDomainRequest{
+		Name:              listDomainsResp.Domains[0].GetDomainInfo().GetName(),
+		ActiveClusterName: &preferredCluster,
+	}
+	s.serverFrontendClient.EXPECT().
+		UpdateDomain(gomock.Any(), expectedUpdateRequest).
+		Return(&types.UpdateDomainResponse{}, nil).Times(1)
+	success, fail := s.domainCLI.rebalanceDomains(s.cliCtx)
+	s.Equal(1, len(success))
+	s.Equal(0, len(fail))
+}
+
+func (s *domainCommandsSuite) TestRebalanceDomains_NoManagedFailover_Skipped() {
+	listDomainsResp := getListDomainsResponse()
+	preferredCluster := "active"
+	listDomainsResp.Domains[0].DomainInfo.Data = map[string]string{
+		common.DomainDataKeyForManagedFailover:  "false",
+		common.DomainDataKeyForPreferredCluster: preferredCluster,
+	}
+	s.serverFrontendClient.EXPECT().ListDomains(gomock.Any(), gomock.Any()).Return(listDomainsResp, nil)
+
+	s.serverFrontendClient.EXPECT().
+		UpdateDomain(gomock.Any(), gomock.Any()).
+		Return(&types.UpdateDomainResponse{}, nil).Times(0)
+	success, fail := s.domainCLI.rebalanceDomains(s.cliCtx)
+	s.Equal(0, len(success))
+	s.Equal(0, len(fail))
+}
+
+func (s *domainCommandsSuite) TestRebalanceDomains_NoPreferredCluster_Skipped() {
+	listDomainsResp := getListDomainsResponse()
+	listDomainsResp.Domains[0].DomainInfo.Data = map[string]string{
+		common.DomainDataKeyForManagedFailover: "true",
+	}
+	s.serverFrontendClient.EXPECT().ListDomains(gomock.Any(), gomock.Any()).Return(listDomainsResp, nil)
+	s.serverFrontendClient.EXPECT().
+		UpdateDomain(gomock.Any(), gomock.Any()).
+		Return(&types.UpdateDomainResponse{}, nil).Times(0)
```

Comment:
nit: Looks like failure case is not tested? Probably adding a test to return random value (success/fail) for update domain and assert # success + # fail returned by `rebalanceDomains` equals to # of domains should be failovered.